src/webview/html/__tests__/contentHtmlFromPieceDescriptors-test.js

```diff
@@ -0,0 +1,235 @@
+/* @flow strict-local */
+
+import * as eg from '../../../__tests__/lib/exampleData';
+import {
+  HOME_NARROW,
+  streamNarrow,
+  topicNarrow,
+  pmNarrowFromUsersUnsafe,
+  keyFromNarrow,
+  ALL_PRIVATE_NARROW,
+} from '../../../utils/narrow';
+import contentHtmlFromPieceDescriptors from '../contentHtmlFromPieceDescriptors';
+import getHtmlPieceDescriptors from '../../../message/getHtmlPieceDescriptors';
+
+/**
+ * Highlight changes in content-HTML generation.
+ *
+ * Test failures here (which we expect to happen often) will have two
+ * major flavors:
+ *
+ * - Your changes caused different content HTML to be generated from
+ *   the same input (list of messages, backgroundData, etc.). You
+ *   should examine the changes and see if we want them.
+ *   - If they look correct, please follow Jest's instructions to
+ *     update the snapshots, and commit the result.
+ *   - If they don't look correct, look for bugs that were caused or
+ *     revealed by your changes. Please fix them and run the tests
+ *     again.
+ *
+ * - Different input was given on this run of the tests, and naturally
+ *   our code produced different output.
+ *   - This will be the case when extending these tests to increase
+ *     coverage. Thanks for doing that! :)
+ *   - But one big "gotcha!" is that we can't vary the input
```

Comment:
Is there another sub-point to go here? Representative values for our model data might have changed; e.g., if Gravatar suddenly told us to change the URL we use to request images, and we did that.